clients/shared/merge.go

```diff
@@ -0,0 +1,164 @@
+package shared
+
+import (
+	"fmt"
+	"log/slog"
+	"time"
+
+	"github.com/artie-labs/transfer/lib/jitter"
+
+	"github.com/artie-labs/transfer/lib/config"
+	"github.com/artie-labs/transfer/lib/config/constants"
+	"github.com/artie-labs/transfer/lib/destination"
+	"github.com/artie-labs/transfer/lib/destination/ddl"
+	"github.com/artie-labs/transfer/lib/destination/dml"
+	"github.com/artie-labs/transfer/lib/destination/types"
+	"github.com/artie-labs/transfer/lib/optimization"
+	"github.com/artie-labs/transfer/lib/ptr"
+	"github.com/artie-labs/transfer/lib/sql"
+	"github.com/artie-labs/transfer/lib/typing/columns"
+)
+
+func Merge(dwh destination.DataWarehouse, tableData *optimization.TableData, cfg config.Config, opts types.MergeOpts) error {
+	if tableData.ShouldSkipUpdate() {
+		return nil
+	}
+
+	fqName := tableData.ToFqName(dwh.Label(), true, cfg.SharedDestinationConfig.UppercaseEscapedNames, "")
```

Comment:
In the prior code the `fqName` for BigQuery included `s.config.BigQuery.ProjectID`

Reply:
I've been thinking about creating a function on the interface for `FqName`, this is a good reason to revive it.